hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/bulk/BulkInsertWriterHelper.java

```diff
@@ -157,9 +165,25 @@ private HoodieRowDataCreateHandle getRowCreateHandle(String partitionPath) throw
   }
 
   public void close() throws IOException {
-    for (HoodieRowDataCreateHandle rowCreateHandle : handles.values()) {
-      LOG.info("Closing bulk insert file " + rowCreateHandle.getFileName());
-      writeStatusList.add(closeWriteHandle(rowCreateHandle));
+    ExecutorService executorService = Executors.newFixedThreadPool(handles.size());
+    allOf(handles.values().stream()
+        .map(rowCreateHandle -> CompletableFuture.supplyAsync(() -> {
+          try {
+            LOG.info("Closing bulk insert file " + rowCreateHandle.getFileName());
+            return rowCreateHandle.close();
+          } catch (IOException e) {
+            throw new HoodieIOException("IOE during rowCreateHandle.close()", e);
+          }
+        }, executorService))
+        .collect(Collectors.toList())
+    ).whenComplete((result, throwable) -> {
+      writeStatusList.addAll(result);
+    }).join();
+    try {
+      executorService.shutdown();
+      executorService.awaitTermination(24, TimeUnit.DAYS);
```

Comment:
24 days?

Reply:
Sorry! 24 hours.  this indicates a long timeout , aiming to waiting for all handles closed.

Reply:
Still too long, the close should be fast right?

Reply:
> Still too long, the close should be fast right?

I agree, close should be fast.  here awaitTermination is actually a timeout, 24 Hours is too long , how about wait for a Flink CP interval?

Reply:
Did you estimate the normal time cost for the close? Maybe just 1 minute should be enough.

Reply:
1 minute maybe not enough when HDFS is busy.  or we can make this a config and set default value 1 minute?

Reply:
Teacher Danny, PTAL @danny0405